### PR TITLE
gradle: cleanup

### PR DESCRIPTION
**Pull request by postamar**
This PR follows up on https://github.com/airbytehq/airbyte/pull/30011 and cleans up the gradle files in a similar fashion. I validated these changes manually by making sure that the targets which are called by the CI or by the scripts in tools or whatever are not more broken than they currently already are.

### CODE REVIEW

File: buildSrc/src/main/groovy/airbyte-python.gradle
Comment (postamar) [inline buildSrc/src/main/groovy/airbyte-python.gradle:109]: This takes care of https://airbytehq-team.slack.com/archives/C02U2SSHP9S/p1669756536254519
cc @erohmensing @rodireich

Comment (aaronsteers) [reply buildSrc/src/main/groovy/airbyte-python.gradle:109]: Nice.

File: buildSrc/src/main/groovy/airbyte-python.gradle
Comment (aaronsteers) [inline buildSrc/src/main/groovy/airbyte-python.gradle:99]: Nice.

File: buildSrc/src/main/groovy/airbyte-python.gradle
Comment (aaronsteers) [inline buildSrc/src/main/groovy/airbyte-python.gradle:118]: @postamar - For my info, post merge, what’s the correct/best entrypoint for python (or generic) linting and auto-formatting?

**Review by aaronsteers — APPROVED** [review]
I reviewed the full PR. No issues found, just one question inline.

Would be good to have one more approver with more familiarity on the build processes. 

Maybe @colesnodgrass - could you also review if you have a sec, since we were just talking about Gradle? 😅

File: buildSrc/src/main/groovy/airbyte-python.gradle
Comment (postamar) [reply buildSrc/src/main/groovy/airbyte-python.gradle:118]: `:format` from the repo root

**Comment by postamar** [comment]
cc @airbytehq/connector-operations as well for 👀

**Review by colesnodgrass — COMMENTED** [review]
Changes look fine.  We should move the gradle task creation and depends-on look-ups to be lazily-evaluated.

E.g. registering a task changes from 
```
task downloadSpecSecretMask(Download) {
``` 
to 
```
tasks.register("downloadSpecSecretMask", Download) {
```

depending on a different task changes from
```
processResources.dependsOn(downloadSpecSecretMask)
```
to
```
tasks.named("processResources") {
  dependsOn tasks.named("downloadSpecSecretMask")
}
```

File: buildSrc/src/main/groovy/airbyte-python.gradle
Comment (erohmensing) [reply buildSrc/src/main/groovy/airbyte-python.gradle:109]: Nice!

**Comment by postamar** [comment]
@colesnodgrass I did as you suggested. The resulting changes are quite extensive and warrant another round of review. I did some manual spot-checking to verify that everything still works, but of course I probably did break something in some corner somewhere. Consequently I'll wait until next week to merge.

File: airbyte-api/build.gradle
Comment (postamar) [inline airbyte-api/build.gradle:202]: I now define a per-project `generate` task in the root `gradle.build`. Its purpose is to group all codegen-related tasks.

File: airbyte-cdk/python/build.gradle
Comment (postamar) [inline airbyte-cdk/python/build.gradle:35]: These tasks are made redundant by `:format`. The last task shouldn't exist as a gradle task imo, instead developers should call the shell script directly if they want.

File: airbyte-ci/connectors/pipelines/pipelines/actions/environments.py
Comment (postamar) [inline airbyte-ci/connectors/pipelines/pipelines/actions/environments.py:532]: We don't actually use gradle for publishing anything but locally anymore.

File: airbyte-ci/connectors/pipelines/pipelines/tests/java_connectors.py
Comment (postamar) [inline airbyte-ci/connectors/pipelines/pipelines/tests/java_connectors.py:56]: The base gradle plugin defines `check` as the umbrella task for testing, `test` is java-specific. TIL.

File: airbyte-commons-cli/build.gradle
Comment (postamar) [inline airbyte-commons-cli/build.gradle:9]: These don't get called by the CI anymore.

File: airbyte-commons/build.gradle
Comment (postamar) [inline airbyte-commons/build.gradle:19]: This was moved here from elsewhere.

File: airbyte-integrations/bases/base-normalization/build.gradle
Comment (postamar) [inline airbyte-integrations/bases/base-normalization/build.gradle:13]: This is because we no longer apply the `java` plugin blindly everywhere. See `build.gradle`.

File: airbyte-integrations/connector-templates/generator/build.gradle
Comment (postamar) [inline airbyte-integrations/connector-templates/generator/build.gradle:33]: I checked: `manage.sh` just delegates to `generate.sh`.

File: build.gradle
Comment (postamar) [inline build.gradle:403]: I added this line.

File: buildSrc/src/main/groovy/airbyte-python.gradle
Comment (postamar) [inline buildSrc/src/main/groovy/airbyte-python.gradle:44]: This function is the same, I just made it a bit more readable.

File: buildSrc/src/main/groovy/airbyte-python.gradle
Comment (postamar) [inline buildSrc/src/main/groovy/airbyte-python.gradle:233]: Let's run python tests using `check` now instead of `test`.

**Comment by octavia-approvington** [comment]
Our work here is done <br/> done

**Review by octavia-approvington — APPROVED** [review]

**Pull request merged**
